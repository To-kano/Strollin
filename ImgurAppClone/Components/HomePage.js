import React, {useState} from 'react';
import {StyleSheet ,Text , View, Image, TextInput, TouchableOpacity} from "react-native";

function HomePage(props) {

  const [value, onChangeText] = React.useState('');

  return (
    <View style={styles.back}>
      <TouchableOpacity
        style={styles.button}
        onPress={()=> console.log("ok")}
      >
        <Image source={require('../ressources/trip2.jpg')} style={styles.buttonImage} />
        <Text> New Trip </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.button}
        onPress={()=> console.log("ok")}
      >
        <Image source={require('../ressources/profil.png')} style={styles.buttonImage} />
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.button}
        onPress={()=> console.log("ok")}
      >
        <Image source={require('../ressources/trip2.jpg')} style={styles.buttonImage} />
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  back: {
      backgroundColor: 'pink',
      flexDirection: 'column',
      justifyContent: 'flex-start',
      alignItems: 'center',
      flex: 1
  },
  buttonImage: {
    width: '100%',
    height: '100%',
    resizeMode: 'cover',
    position: 'absolute'
  },
  button: {
    alignItems: "center",
    backgroundColor: "#DDDDDD",
    justifyContent: 'center',
    padding: 10,
    flex: 1,
    flexDirection: 'column',
    width: '100%',
    marginBottom: 10,
    marginTop: 10
  }
});

export default HomePage;
