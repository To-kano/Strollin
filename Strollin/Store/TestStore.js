import { createStore, combineReducers } from 'redux';
import TestReducer from './Reducers/testReducer';

const rootReducer = combineReducers({
    test: TestReducer
});

export default createStore(rootReducer);
