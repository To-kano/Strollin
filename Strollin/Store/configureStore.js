import { createStore, combineReducers } from 'redux';
import profileReducer from './Reducers/profileReducer';
import galleryReducer from './Reducers/galleryReducer';
import geolocalisationReducer from './Reducers/geolocalisationReducer';
import mapNavigationReducer from './Reducers/mapNavigationReducer';
import chatConversation from './Reducers/conversationReducer';
import socketReducer from './Reducers/socketReducer';
import tendanceReducer from './Reducers/tendanceReducer';
import messageReducer from './Reducers/messageReducer';

const rootReducer = combineReducers({
  profil: profileReducer,
  gallery: galleryReducer,
  position: geolocalisationReducer,
  map: mapNavigationReducer,
  conversation: chatConversation,
  socket: socketReducer,
  tendance: tendanceReducer,
  message: messageReducer
});

export default createStore(rootReducer);
