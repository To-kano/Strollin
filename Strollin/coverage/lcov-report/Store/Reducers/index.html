
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for Store/Reducers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> Store/Reducers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">25.33% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>38/150</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">34.48% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>20/58</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.63% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/19</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">25.33% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>38/150</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="conversationReducer.js"><a href="conversationReducer.js.html">conversationReducer.js</a></td>
	<td data-value="20" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 20%"></div><div class="cover-empty" style="width: 80%"></div></div>
	</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="20" class="abs low">4/20</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="8" class="abs low">2/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="20" class="abs low">4/20</td>
	</tr>

<tr>
	<td class="file low" data-value="createConversationReducer.js"><a href="createConversationReducer.js.html">createConversationReducer.js</a></td>
	<td data-value="26.67" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 26%"></div><div class="cover-empty" style="width: 74%"></div></div>
	</td>
	<td data-value="26.67" class="pct low">26.67%</td>
	<td data-value="15" class="abs low">4/15</td>
	<td data-value="28.57" class="pct low">28.57%</td>
	<td data-value="7" class="abs low">2/7</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="26.67" class="pct low">26.67%</td>
	<td data-value="15" class="abs low">4/15</td>
	</tr>

<tr>
	<td class="file medium" data-value="galleryReducer.js"><a href="galleryReducer.js.html">galleryReducer.js</a></td>
	<td data-value="60" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 60%"></div><div class="cover-empty" style="width: 40%"></div></div>
	</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="5" class="abs medium">3/5</td>
	</tr>

<tr>
	<td class="file low" data-value="geolocalisationReducer.js"><a href="geolocalisationReducer.js.html">geolocalisationReducer.js</a></td>
	<td data-value="27.27" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 27%"></div><div class="cover-empty" style="width: 73%"></div></div>
	</td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="6" class="abs low">2/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	</tr>

<tr>
	<td class="file low" data-value="mapNavigationReducer.js"><a href="mapNavigationReducer.js.html">mapNavigationReducer.js</a></td>
	<td data-value="9.68" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 9%"></div><div class="cover-empty" style="width: 91%"></div></div>
	</td>
	<td data-value="9.68" class="pct low">9.68%</td>
	<td data-value="31" class="abs low">3/31</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="7" class="abs low">1/7</td>
	<td data-value="9.68" class="pct low">9.68%</td>
	<td data-value="31" class="abs low">3/31</td>
	</tr>

<tr>
	<td class="file medium" data-value="messageReducer.js"><a href="messageReducer.js.html">messageReducer.js</a></td>
	<td data-value="62.5" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 62%"></div><div class="cover-empty" style="width: 38%"></div></div>
	</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	</tr>

<tr>
	<td class="file low" data-value="profileReducer.js"><a href="profileReducer.js.html">profileReducer.js</a></td>
	<td data-value="11.43" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 11%"></div><div class="cover-empty" style="width: 89%"></div></div>
	</td>
	<td data-value="11.43" class="pct low">11.43%</td>
	<td data-value="35" class="abs low">4/35</td>
	<td data-value="14.29" class="pct low">14.29%</td>
	<td data-value="14" class="abs low">2/14</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="11.43" class="pct low">11.43%</td>
	<td data-value="35" class="abs low">4/35</td>
	</tr>

<tr>
	<td class="file low" data-value="searchReducer.js"><a href="searchReducer.js.html">searchReducer.js</a></td>
	<td data-value="36.36" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 36%"></div><div class="cover-empty" style="width: 64%"></div></div>
	</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	<td data-value="40" class="pct low">40%</td>
	<td data-value="5" class="abs low">2/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="36.36" class="pct low">36.36%</td>
	<td data-value="11" class="abs low">4/11</td>
	</tr>

<tr>
	<td class="file medium" data-value="socketReducer.js"><a href="socketReducer.js.html">socketReducer.js</a></td>
	<td data-value="66.67" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 66%"></div><div class="cover-empty" style="width: 34%"></div></div>
	</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="6" class="abs medium">4/6</td>
	</tr>

<tr>
	<td class="file medium" data-value="tendanceReducer.js"><a href="tendanceReducer.js.html">tendanceReducer.js</a></td>
	<td data-value="50" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 50%"></div><div class="cover-empty" style="width: 50%"></div></div>
	</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Mon Jan 04 2021 15:58:26 GMT+0100 (Central European Standard Time)
            </div>
        </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    