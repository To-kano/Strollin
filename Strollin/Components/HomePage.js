import React, { useState } from 'react';
import {
  StyleSheet, Text, View, Image, TouchableOpacity, FlatList, Linking
} from 'react-native';
import { connect } from 'react-redux';
import CourseItem from './CourseItem';
import I18n from '../Translation/configureTrans';
import SearchBar from './TendanceSearchBar';
import Store from '../Store/configureStore';

import DrawerLayout from 'react-native-gesture-handler/DrawerLayout';
import Menu from './Menu';

import { DrawerActions } from '@react-navigation/native';


const imageFriend = require('../ressources/friend.png');
// const imageHistory = require('../ressources/history.png');
const imageProfile = require('../ressources/profile.png');

export function setSortedTendanceData(tag) {
  const store = Store.getState();
  const sortedData = [];

  console.log("tag = ", tag)
  for (i in store.tendance.tendanceList) {
    for (j in store.tendance.tendanceList[i].tags_list) {
      if (store.tendance.tendanceList[i].tags_list[j] == tag) {
        sortedData.push(store.tendance.tendanceList[i]);
        break;
      }
    }
  }
  // console.log("sortedData = ", sortedData);
  const action = {
    type: 'SET_SORTED_TENDANCE_LIST',
    value: sortedData
  };
  Store.dispatch(action);
}

export function getTendanceList() {
  const store = Store.getState();

  if (store.tendance.sortedTendanceList.length > 0) {
    return (store.tendance.sortedTendanceList);
  }
  return (store.tendance.tendanceList);
}

function getUrl(props) {
  console.log(props)
  if (props.profil.partner === true)
    return ("https://forms.gle/feBUmJ8wQFPGga4A6")
  return ('https://forms.gle/cR126bgPaexbgGYD9')
}

export function Header({ props, defaultState = false }) {
  const [pressed, setpressed] = useState(defaultState);
  let url = getUrl(props)
  if (pressed === false) {
    return (
      <View style={styles.view_header}>
        <TouchableOpacity
          onPress={() => props.navigation.dispatch(DrawerActions.openDrawer())}
        >
          <Image style={styles.img_header} source={require('../images/icons/black/menu.png')} />
        </TouchableOpacity>
        <Text style={styles.text_header}>
          {I18n.t('Header.home')}
        </Text>
        <TouchableOpacity
          onPress={() => { Linking.openURL(url) }}
        >
          <Image style={styles.img_header} source={require('../images/icons/black/form.png')} />
        </TouchableOpacity>
        <TouchableOpacity
          onPress={() => { setpressed(!pressed); }}
        >
          <Image style={styles.img_header} source={require('../images/icons/black/search.png')} />
        </TouchableOpacity>
      </View>
    );
  }
  return (
    <View style={styles.view_header}>
      <SearchBar
        onPress={(data) => { setSortedTendanceData(data); setpressed(!pressed); }}
        imagePath="../images/icons/black/search.png"
      />
    </View>
  );
}

export function HomePage(props) {

  return (
    <View style={styles.view_back}>
      <Header props={props} />
      <View style={styles.view_list}>
        <FlatList
          showsHorizontalScrollIndicator={false}
          showsVerticalScrollIndicator={false}
          data={getTendanceList()}
          renderItem={({ item }) => (
            <CourseItem
              {...props}
              data={item}
            />
          )}
          keyExtractor={(item) => item.id}
        />
        {/* export function HomePage(props) {

  const [drawer, setDrawer] = useState(null);

  return (
    <View style={{ flex: 1 }}>
        <View style={styles.view_back}>
          <View style={styles.view_header}>
            <TouchableOpacity onPress={//() => props.navigation.navigate('Menu')
             () => props.navigation.dispatch(DrawerActions.openDrawer())
            }>
              <Image style={styles.img_header} source={require('../images/icons/black/menu.png')}/>
            </TouchableOpacity>
            <Text style={styles.text_header}>Home</Text>
            <TouchableOpacity>
              <Image style={styles.img_header} source={require('../images/icons/black/search.png')}/>
            </TouchableOpacity>
          </View>
          <View style={styles.SearchBar}>
              <SearchBar
               onPress={setSortedTendanceData}
               imagePath='../images/icons/black/search.png'
              />
          </View>
          <View style={styles.viex_list}>
            <FlatList
              showsHorizontalScrollIndicator={false}
              showsVerticalScrollIndicator={false}
              data={getTendanceList()}
              renderItem={({ item }) => (
                <TendenceCourseItem
                  {...props}
                  data={item}
                />
              )}
              keyExtractor={(item) => item["_id"]}
            />
          </View> */}

      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  view_back: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    alignItems: 'center',
    backgroundColor: '#E1E2E7',
    paddingTop: '1.8%',
    paddingLeft: '3.3%',
    paddingRight: '3.3%',
    paddingBottom: '0%',
  },
  view_header: {
    flex: 50,
    flexDirection: 'row',
    alignItems: 'center',
    marginBottom: 10,
  },
  textInput_header: {
    height: 40,
    width: '85%',
    borderRadius: 21,
    marginRight: 12.5,
    paddingLeft: 12.5,
    backgroundColor: '#FFFFFF',
  },
  img_header: {
    width: 34,
    resizeMode: 'contain',
  },
  text_header: {
    width: '77.8%',
    fontWeight: 'bold',
    fontSize: 28,
    letterSpacing: 2,
    textAlign: 'center',
    color: '#000000',
  },
  view_list: {
    flex: 757,
    width: '100%',
  }
});

const mapStateToProps = (state) => state;
export default connect(mapStateToProps)(HomePage);
