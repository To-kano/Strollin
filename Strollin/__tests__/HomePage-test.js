import React from 'react';
import {shallow} from 'enzyme';
import HomePage from '../Components/HomePage';

import Store from '../Store/configureStore';
import { Provider } from 'react-redux';

const navigationTest = {
    navigate: (test) => jest.fn()
}

const dispatchTest = (test) => jest.fn()

describe('HomePage', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(
            <Provider store={Store}>
                <HomePage navigation={navigationTest} dispatch={dispatchTest} />
            </Provider>
            )
            expect(component).toMatchSnapshot()
        });
    });
});