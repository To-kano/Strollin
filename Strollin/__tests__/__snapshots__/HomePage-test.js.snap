// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
